hack/dind-cluster.sh

```diff
@@ -125,6 +125,7 @@ function start() {
 
   if [[ -n "${ADDITIONAL_NETWORK_INTERFACE}" ]]; then
     add-network-interface-to-nodes
+    update-node-config
```

Comment:
I fail to understand why we need to remove the file first?

Reply:
The idea here is to reuse exiting  ensure-node-config() in images/dind/node/openshift-generate-node-config.sh for generating config/certs (avoids code duplication).

openshift-generate-node-config.sh is called every time when the openshift-node service is started: https://github.com/openshift/origin/blob/master/images/dind/node/openshift-node.service#L7. 

ensure-node-config() will not regenerate certs if the node-config.yaml already exists: https://github.com/openshift/origin/blob/master/images/dind/node/openshift-generate-node-config.sh#L26
So removing the existing config file will trigger ensure-node-config() to generate fresh certs/config.

Reply:
Would it simplify things to just always generate two interfaces rather than having the "-a" flag? I don't think there's any reason you'd ever need to *avoid* having a second interface. And I'm not sure there's much use case for having more than two interfaces...

Also, what's the argument for having two bridges (one for eth0 and one for everything else) rather than just only having one bridge with everything? (It seems like if there really is a need to keep eth0 and eth1 separate, and there really is a need for being able to have both eth1 and eth2, then there's likely to be a need to keep eth1 and eth2 separate too...)

Reply:
> And I'm not sure there's much use case for having more than two interfaces...
I think we might need one more to be used as a gateway in ovn.

> Also, what's the argument for having two bridges
Don't we need two bridges for keeping the two networks separate?

Reply:
> Don't we need two bridges for keeping the two networks separate?

Define "separate". You can run multiple independent IP subnets on the same "wires". Which (assuming I read this right) is what this code does if you add more than 1 extra interface: eth0 gets its own bridge, eth1 gets a second bridge, but then eth2, eth3, etc all share that second bridge with eth1 rather than getting their own bridges. But if that works for the more-than-2-networks case, then why not use a shared bridge in the exactly-2-networks case too?

Reply:
May be I was not clear. In dind, eth0 on all nodes/master are on one bridge (docker0), similarly additional interface eth1 on all nodes/master will be on another bridge (docker10) with '-a' option. This pr doesn't deal with more than 2 interfaces (no eth2, eth3...). Option '-a' also enables master to listen on eth1 interface (this forces node to talk to master through it's eth1 interface).
If we only need to test traffic going through eth0 or eth1, then we don't care about number of bridges? as Dan suggested both eth0 and eth1 could be on single docker0 bridge?
@rajatchopra @danwinship

Reply:
Oh, I see, I misunderstood the commit message. Before, all the eth0s were on docker0, but each node's eth1 had its own separate dockerN bridge connecting it only to the master. This fixes things so that instead you have docker0 for all the eth0s and docker10 for all the eth1s. Got it.